app/javascript/dashboard/store/modules/reports.js

```diff
@@ -120,7 +120,24 @@ export const actions = {
 
         document.body.appendChild(downloadLink);
         downloadLink.click();
-        // document.body.removeChild(downloadLink);
+        document.body.removeChild(downloadLink);
+      })
+      .catch(error => {
+        console.error(error);
+      });
+  },
+  downloadTeamReports(_, reportObj) {
+    return Report.getTeamReports(reportObj.from, reportObj.to)
+      .then(response => {
+        let csvContent = 'data:text/csv;charset=utf-8,' + response.data;
+        var encodedUri = encodeURI(csvContent);
+        var downloadLink = document.createElement('a');
+        downloadLink.href = encodedUri;
+        downloadLink.download = reportObj.fileName;
+
+        document.body.appendChild(downloadLink);
+        downloadLink.click();
+        document.body.removeChild(downloadLink);
       })
```

Comment:
This is repeated in the function above. Can you please create a common method and write a spec for it?

Reply:
Resolved in https://github.com/chatwoot/chatwoot/pull/3116/commits/2e9641d9a5ee2d671424162eae37474b7aa4a6c6